### DISCUSSION TITLE
using javascript-tide before javascript-eslint

### DISCUSSION THREAD

**Discussion by Aaronzinhoo** [ananthakumaran/tide#422] [Q&A]
Hey, so I have been using `rjsx-mode` for some time now and since I recently learned of the tide checkers I wanted to add javascript tide to its checkers. I was able to successfully load it, but every time it loads, `javascript-eslint` is loaded before it. Is there any way to change the ordering so that `javascript-tide` is preferred or loaded before `javascript-eslint`?

```
:config
  (flycheck-add-mode 'javascript-eslint 'rjsx-mode)
  (flycheck-add-next-checker 'jsx-tide '(warning . javascript-eslint) 'append)
  (flycheck-add-next-checker 'javascript-tide '(warning . javascript-eslint) 'append)
  (flycheck-add-mode 'typescript-tide 'ng2-ts-mode)
  (flycheck-add-mode 'javascript-eslint 'ng2-ts-mode)
  (flycheck-add-mode 'typescript-tslint 'ng2-ts-mode)
  (flycheck-add-next-checker 'typescript-tide '(warning . javascript-eslint) 'append)
  (if (file-exists-p (concat tide-project-root "node_modules/typescript/bin/tsserver"))
      (setq tide-tsserver-executable "node_modules/typescript/bin/tsserver"))
```

This is the relevant portion of my config. I also modified the `javascript-tide` modes it activates for. Any suggesitons?

I got it to work by creating a hook function and disabling the add-next-checker line. My `tide-mode` hook simply starts the tide server and sets some presets I like for my dev environment. I do not think this is a good solution but it works at least. If you know a better method I would be open to suggestions.
```
(defun aaronzinhoo-tide-rjsx-mode-hook ()
    (aaronzinhoo-tide-mode-hook)
    (flycheck-select-checker 'javascript-tide))
```

**Comment by ananthakumaran** [comment]
I don't see an officially documented approach to set the next checkers https://github.com/flycheck/flycheck/issues/836 https://github.com/flycheck/flycheck/issues/1660

The way you do currently seems ok to me. One thing to note is, we define multiple tide checkers `{mode-name}-tide`.  For rjsx mode, jsx-tide should be used.

```
// after flycheck and tide is loaded
(flycheck-add-next-checker 'jsx-tide '(warning . javascript-eslint))

// in your rjsx mode hook
(setq flycheck-checker 'jsx-tide)
```
